### ISSUE TITLE
ARIA Issues with Dropdown Menu

### ISSUE THREAD

**Issue by jmmadsen** [ericgio/react-bootstrap-typeahead#517]
### Steps to reproduce

Using the AsyncTypeahead component, I am getting serious/critical issues warning from the `react-axe` accessibility package, which logs issues to the console. 

![image](https://user-images.githubusercontent.com/57506409/74172175-c6107d80-4bfd-11ea-87d0-c59a6dcec527.png)

![image](https://user-images.githubusercontent.com/57506409/74172189-cdd02200-4bfd-11ea-9e97-abe3035a00a8.png)

### Expected Behavior

I have noticed throughout your documentation and other issues, you place a priority on accessibility. I am not using `renderMenu` prop, so I would expect the accessibility features to work out of the box.

### Actual Behavior

I looked through the documentation, and attempted to use `renderMenu` and `renderMenuItemChildren`, but it does not appear to override the parent element's accessibility issues. I am using `inputProps` to get aria-labels into that component successfully, but this does not appear to work for the rendering props?

Any advice/suggestions on this accessibility issue would be greatly appreciated!

**Comment by ericgio** [comment]
Hi @jmmadsen, and thanks for filing this issue! I do care about accessibility and strive to make this component as compliant with the current standards as possible.

For the first warning, I believe that's just a matter of providing a `name` attribute for the input. While I could provide a default name, I think it makes more sense to leave it up to the consumer to add a name, just as they would with any other input element. You can add a `name` using `inputProps`:

```jsx
<Typeahead
  ...
  inputProps={{
    name: 'my-input-label',
  }}
/>
```

I'll have to take a closer look at the second warning. Both the warning and the [explanatory URL](https://dequeuniversity.com/rules/axe/3.4/aria-required-children) are a bit vague, so it's unclear to me at first glance why that's happening. If you know how to fix the warning, I'd love to hear an explanation. Thanks again!

**Comment by jmmadsen** [comment]
Eric,

Thanks for the quick response.  I tried the name in the `inputProps`, but the warning still appears. I believe the issue is that `inputProps` pertains to the typable search field itself, whereas the warning is coming from the actual rendered menu generated beneath it that lacks a name.

![image](https://user-images.githubusercontent.com/57506409/74177981-c7937300-4c08-11ea-9560-257656d36eb8.png)

As for the second issue, I think it may actually be a problem with react-bootstrap as they do not assign aria roles off the bat and leave it to the developer. I am messing around with the `renderMenu` function now to see if I can get the proper arias in there - in which case, I can share the code and it should be as simple as adding the proper role to the bootstrap elements in the source code.

I'll keep plugging away, and please let me know if you find anything on your end. I appreciate the help!

**Comment by jmmadsen** [comment]
Sharing the component code before my changes for transparency (minus the name in `inputProps` as that did not fix the issue.

```<NavItem className="searchForm-container" aria-label='nav-container-for-receptacle-search-form'>
				<Fragment>
					<AsyncTypeahead
						id="receptacle-id-typeahead-autocomplete-is-on"
						isLoading={isLoading}
						disabled={disabled}
						minLength={minLength}
						labelKey="receptacleSearch"
						options={options}
						placeholder="Search for UPU Barcode"
						maxResults={maxResults}
						onSearch={this._handleSearch}
						onChange={this.handleChange}
						delay={250}
						paginate={false}
						paginationText={"Click to load more results"}
						onPaginate={this.onPaginate}
						selectHintOnEnter={true}
						useCache={false}
						inputProps={{ autoComplete: 'off', "aria-label": 'receptacle-id-autocomplete-is-on'}}
					/>
				</Fragment>
			</NavItem>```

**Comment by jmmadsen** [comment]
@ericgio Any new info on this? I wasn't able to get anywhere, and had to move on to other 508 issues in my application.

**Comment by ericgio** [comment]
I can reproduce both of these issues using the `react-axe` package with the package examples.

**"ARIA input fields have an accessible name"**
Because the menu has `role="listbox"`, it also expects to have a label of some sort:

> Fix any of the following:
> - aria-label attribute does not exist or is empty
> - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
> - Element has no title attribute or the title attribute is empty

Adding a default `aria-label` to the menu will fix the warning and should be acceptable as a solution.

**"Certain ARIA roles must contain particular children"**
This is happening because the menu item displayed during the empty/searching state doesn't have the correct role:

> Required ARIA child role not present: option

The fix here is just to add `role="option"` to the empty label menu item.

**Comment by ericgio** [comment]
Fixed in v4.0.0-rc.2

**Comment by jmmadsen** [comment]
Thank you! Appreciate your help.
